tox.ini

```diff
@@ -2,15 +2,17 @@
 envlist =
     flake8,
     docs,
-    py{36,37,38,39}-dj{32,31,22},
-    py{38,39}-djmain,
+    py{37,38,39}-dj22,
+    py{37,38,39}-dj32,
```

Comment:
Should include 3.10 since dj 3.2.9 added support

Reply:
> @jezdez I think this should be merged though with a change in setup.py to disallow Django 4.0.0 but allow 4.0.1. That way it'd be included in this new minor release as it seems like current maintainers are fairly busy.

That's...not really a great way of specifying dependencies. 

@auvipy Please make sure to coordinate with @n2ygk and @MattBlack85 before tagging. We've talked about this before, you can't just ask for review in https://github.com/jazzband/django-oauth-toolkit/issues/1000#issuecomment-995895750 and then go ahead and create a tag on your own. Drafting a release is one thing, but publishing the release here is another.

> 
> We can just delete the current git tag and re tag since not released on pypi yet

Sure, just know that every git tag triggers the release process and sends an email to the project leads to review it, so coordinating this beforehand would be better.

Reply:
> > @jezdez I think this should be merged though with a change in setup.py to disallow Django 4.0.0 but allow 4.0.1. That way it'd be included in this new minor release as it seems like current maintainers are fairly busy.
> 
> That's...not really a great way of specifying dependencies.
> 
> @auvipy Please make sure to coordinate with @n2ygk and @MattBlack85 before tagging. We've talked about this before, you can't just ask for review in [#1000 (comment)](https://github.com/jazzband/django-oauth-toolkit/issues/1000#issuecomment-995895750) and then go ahead and create a tag on your own. Drafting a release is one thing, but publishing the release here is another.
> 
> > We can just delete the current git tag and re tag since not released on pypi yet
> 
> Sure, just know that every git tag triggers the release process and sends an email to the project leads to review it, so coordinating this beforehand would be better.

I haven't seen any guideline for that, and people was bugging me for drafting a release. they were not available for long time if you check. as the github tag push is for review I thought it is drafting the release. hence the confusion. I think toy should specify loudly to people that not all regular contributors are project lead or able to push e tag. this type of mixed situation in a same or is kind of frustrating & de-motivating. I think it's wise to not involve in any jazzband projects without being a project lead.

Reply:
@jezdez can you recommend a way to pin dependencies? I think allowing django-oauth-toolkit run on Django 4.0.0 will cause the same mass production failures that Django 3.2 brought with default auto field. I was just thinking `django<4.0.0,>=4.0.1` (sorry if that's not the right syntax).

Or are you saying don't pin at all? Can you elaborate since I'm not quite understanding 😅

Reply:
> > > @jezdez I think this should be merged though with a change in setup.py to disallow Django 4.0.0 but allow 4.0.1. That way it'd be included in this new minor release as it seems like current maintainers are fairly busy.
> > 
> > 
> > That's...not really a great way of specifying dependencies.
> > @auvipy Please make sure to coordinate with @n2ygk and @MattBlack85 before tagging. We've talked about this before, you can't just ask for review in [#1000 (comment)](https://github.com/jazzband/django-oauth-toolkit/issues/1000#issuecomment-995895750) and then go ahead and create a tag on your own. Drafting a release is one thing, but publishing the release here is another.
> > > We can just delete the current git tag and re tag since not released on pypi yet
> > 
> > 
> > Sure, just know that every git tag triggers the release process and sends an email to the project leads to review it, so coordinating this beforehand would be better.
> 
> I haven't seen any guideline for that, and people was bugging me for drafting a release. they were not available for long time if you check. as the github tag push is for review I thought it is drafting the release. hence the confusion.

I hear you, can you elaborate what you mean that the project leads haven't been available for a long time? Who was bugging you to do a release?

To clarify what "drafting a release" means: that's for example updating the changelog, reviewing any potentially needed PRs like this one and finally either open a PR asking the project leads to review it or (as you chose) to create the release. I've gotten word from the project lead @n2ygk that he didn't not expect the release and would have happily done code reviews if being requested to do one.

> I think toy should specify loudly to people that not all regular contributors are project lead or able to push e tag. this type of mixed situation in a same or is kind of frustrating & de-motivating. I think it's wise to not involve in any jazzband projects without being a project lead.

But @MattBlack85 and @n2ygk are the project leads for this project, I'm not sure I follow your argument.

Reply:
> @jezdez can you recommend a way to pin dependencies? I think allowing django-oauth-toolkit run on Django 4.0.0 will cause the same mass production failures that Django 3.2 brought with default auto field. I was just thinking `django<4.0.0,>=4.0.1` (sorry if that's not the right syntax).
> 
> Or are you saying don't pin at all? Can you elaborate since I'm not quite understanding 😅

I'm not sure I follow, is this about https://github.com/django/django/pull/15205? If Django 4.0.0 isn't possible to run properly, then it should be excluded with the [dependency spec](https://www.python.org/dev/peps/pep-0508/), with the `!=4.0.0` notation. That will prevent the installation of 4.0.0 at least and would create a installation error for a user who would pin to Django 4.0.0 in one of their other project dependencies.

Reply:
@jezdez it was in response to the quote below, and yes your suggestion was what I was thinking as a way to avoid installation of Django 4.0.0 and django-oauth-toolkit.

>> @jezdez I think this should be merged though with a change in setup.py to disallow Django 4.0.0 but allow 4.0.1. That way it'd be included in this new minor release as it seems like current maintainers are fairly busy.
> That's...not really a great way of specifying dependencies.

Reply:
@Andrew-Chen-Wang I've pushed a commit that makes the minimum dj40 be 4.0.1 but I see from your comments that there are other concerns that may make this not yet ready to merge? I'll mark this PR WIP to make sure it doesn't get merged just yet.

Reply:
@n2ygk the docs should now be updated. I've also added the !=4.0.0 constraint to setup.cfg to prevent installation. Let me know if that's not a good practice.